### PR TITLE
executor: migrate test-infra to testify for metrics_reader_test.go

### LINKED ISSUE #28598: migrate test-infra to testify for `metrics_reader_test.go`

### PR DESCRIPTION
**Pull request by feitian124**
### What problem does this PR solve?

Issue Number: close #28598 

```release-note
None
```

### CODE REVIEW

File: executor/metrics_reader_test.go [hunk 1]
```diff
@@ -60,13 +63,13 @@ func (s *testSuite7) TestStmtLabel(c *C) {
...
 		preprocessorReturn := &plannercore.PreprocessorReturn{}
-		err = plannercore.Preprocess(tk.Se, stmtNode, plannercore.WithPreprocessorReturn(preprocessorReturn))
-		c.Check(err, IsNil)
-		c.Assert(err, IsNil)
-		_, _, err = planner.Optimize(context.TODO(), tk.Se, stmtNode, preprocessorReturn.InfoSchema)
-		c.Assert(err, IsNil)
-		c.Assert(executor.GetStmtLabel(stmtNode), Equals, tt.label)
+		err = plannercore.Preprocess(tk.Session(), stmtNode, plannercore.WithPreprocessorReturn(preprocessorReturn))
+		require.NoError(t, err)
+		require.NoError(t, err)
```
Comment (mjonss) [inline executor/metrics_reader_test.go:70]:
```suggestion
```

**Review by mjonss — APPROVED** [review]
LGTM with a minor suggestion (Check + Assert => only one require.NoError)

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**